src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -584,10 +582,7 @@ public function overrideGlobals()
      */
     public static function setTrustedProxies(array $proxies, int $trustedHeaderSet)
     {
-        if (self::HEADER_X_FORWARDED_ALL === $trustedHeaderSet) {
-            trigger_deprecation('symfony/http-foundation', '5.2', 'The "HEADER_X_FORWARDED_ALL" constant is deprecated, use either "HEADER_X_FORWARDED_FOR | HEADER_X_FORWARDED_HOST | HEADER_X_FORWARDED_PORT | HEADER_X_FORWARDED_PROTO" or "HEADER_X_FORWARDED_AWS_ELB" or "HEADER_X_FORWARDED_TRAEFIK" constants instead.');
-        }
-        self::$trustedProxies = array_reduce($proxies, function ($proxies, $proxy) {
+        self::$trustedProxies = array_reduce($proxies, static function ($proxies, $proxy) {
```

Comment:
reverting as this might create needles merge conflicts and is not needed: the surrounding method is already static, so is this closure.